packages/expo-in-app-purchases/ios/EXInAppPurchases/EXInAppPurchasesModule.m

```diff
@@ -0,0 +1,449 @@
+    _promises[key] = @[resolve, reject];
+    return YES;
+  } else {
+    reject(@"E_UNFINISHED_PROMISE", @"Must wait for promise to resolve before recalling function.", nil);
+    return NO;
+  }
+}
+
+- (void)resolvePromise:(NSString*)key value:(id)value
+{
+  NSArray *currentPromise = _promises[key];
+  
+  if (currentPromise != nil) {
+    UMPromiseResolveBlock resolve = currentPromise[0];
+    _promises[key] = nil;
+    
+    resolve(value);
+  }
+}
+
+- (void)rejectPromise:(NSString*)key code:(NSString*)code message:(NSString*)message error:(NSError*) error
+{
+  NSArray* currentPromise = _promises[key];
+  
+  if (currentPromise != nil) {
+    UMPromiseRejectBlock reject = currentPromise[1];
+    _promises[key] = nil;
+    
+    reject(code, message, error);
+  }
+}
+
+# pragma mark - Formatting Response
+
+- (NSDictionary *)getProductData:(SKProduct *)product
+{
+  // Use with caution: P0D also implies non-renewable subscription.
+  NSString *subscriptionPeriod = [self getSubscriptionPeriod:product];
+  NSString *type = [subscriptionPeriod isEqualToString:kEXInAppSubPeriod] ? kEXInApp : kEXSubs;
+  
+  NSDecimalNumber *oneMillion = [[NSDecimalNumber alloc] initWithInt:1000000];
+  NSDecimalNumber *priceAmountMicros = [product.price decimalNumberByMultiplyingBy:oneMillion];
+  NSString *price = [NSString stringWithFormat:@"%@%@", product.priceLocale.currencySymbol, product.price];
+  
+  return @{
+           @"description": product.localizedDescription,
+           @"price": price,
+           @"priceAmountMicros": priceAmountMicros,
+           @"priceCurrencyCode": product.priceLocale.currencyCode,
+           @"productId": product.productIdentifier,
+           @"subscriptionPeriod": subscriptionPeriod,
+           @"title": product.localizedTitle,
+           @"type": type
+           };
+}
+
+- (NSDictionary *)getTransactionData:(SKPaymentTransaction *)transaction acknowledged:(BOOL)acknowledged
+{
+  NSData *receiptData = [NSData dataWithContentsOfURL:[[NSBundle mainBundle] appStoreReceiptURL]];
+  
+  // Get original transaction info if any
+  SKPaymentTransaction *originalTransaction = transaction.originalTransaction;
+  NSString *originalTransactionId = originalTransaction ? originalTransaction.transactionIdentifier : NSString.string;
+  NSNumber *originalTransactionTime = originalTransaction ?
+  @(originalTransaction.transactionDate.timeIntervalSince1970 * 1000) : @(0);
+  
+  return @{
+           @"acknowledged": @(acknowledged),
+           @"productId": transaction.payment.productIdentifier,
+           @"purchaseToken": transaction.transactionIdentifier,
+           @"purchaseState": @(transaction.transactionState),
+           @"purchaseTime": @(transaction.transactionDate.timeIntervalSince1970 * 1000),
+           @"transactionReceipt": [receiptData base64EncodedStringWithOptions:0],
+           @"originalPurchaseTime": originalTransactionTime,
+           @"originalPurchaseToken": originalTransactionId
```

Comment:
The name is a bit misleading on that one - `originalPurchaseToken` is not a token, but an ID (apple uses simple large numbers), not to be confused with android's `linkedPurchaseToken`

Reply:
Resolved in https://github.com/expo/expo/pull/4416/commits/d3cf1a0776fd0438d879d301f56294d40c0e50e1